tests/std/tests/GH_001596_adl_proof_algorithms/test.compile.pass.cpp

```diff
@@ -0,0 +1,236 @@
+    std::for_each(ExecutionPolicy, varr, varr, validating_truth{});
+    std::for_each(ExecutionPolicy, iarr, iarr, validating_truth{});
+
+    (void) std::for_each_n(ExecutionPolicy, varr, 0, simple_truth{});
+    (void) std::for_each_n(ExecutionPolicy, varr, 0, validating_truth{});
+    (void) std::for_each_n(ExecutionPolicy, iarr, 0, validating_truth{});
+
+    (void) std::find(ExecutionPolicy, varr, varr, validator{});
+
+    (void) std::find_if(ExecutionPolicy, varr, varr, simple_truth{});
+    (void) std::find_if(ExecutionPolicy, iarr, iarr, validating_truth{});
+
+    (void) std::find_if_not(ExecutionPolicy, varr, varr, simple_truth{});
+    (void) std::find_if_not(ExecutionPolicy, iarr, iarr, validating_truth{});
+
+    (void) std::find_end(ExecutionPolicy, varr, varr, varr, varr);
+    (void) std::find_end(ExecutionPolicy, varr, varr, varr, varr, validating_equal{});
+    (void) std::find_end(ExecutionPolicy, iarr, iarr, iarr, iarr, validating_equal{});
+
+    (void) std::find_first_of(ExecutionPolicy, varr, varr, varr, varr);
+    (void) std::find_first_of(ExecutionPolicy, varr, varr, varr, varr, validating_equal{});
+    (void) std::find_first_of(ExecutionPolicy, iarr, iarr, iarr, iarr, validating_equal{});
+
+    (void) std::adjacent_find(ExecutionPolicy, varr, varr);
+    (void) std::adjacent_find(ExecutionPolicy, iarr, iarr, validating_equal{});
+
+    (void) std::count(ExecutionPolicy, varr, varr, validator{});
+
+    (void) std::count_if(ExecutionPolicy, varr, varr, simple_truth{});
+    (void) std::count_if(ExecutionPolicy, iarr, iarr, validating_truth{});
+
+    (void) std::mismatch(ExecutionPolicy, varr, varr, varr);
+    (void) std::mismatch(ExecutionPolicy, varr, varr, varr, varr);
+    (void) std::mismatch(ExecutionPolicy, iarr, iarr, iarr, validating_equal{});
+    (void) std::mismatch(ExecutionPolicy, iarr, iarr, iarr, iarr, validating_equal{});
+
+    (void) std::equal(ExecutionPolicy, varr, varr, varr);
+    (void) std::equal(ExecutionPolicy, varr, varr, varr, varr);
+    (void) std::equal(ExecutionPolicy, iarr, iarr, iarr, validating_equal{});
+    (void) std::equal(ExecutionPolicy, iarr, iarr, iarr, iarr, validating_equal{});
+
+    (void) std::search(ExecutionPolicy, varr, varr, varr, varr);
+    (void) std::search(ExecutionPolicy, iarr, iarr, iarr, iarr, validating_equal{});
+
+    (void) std::search_n(ExecutionPolicy, varr, varr, 0, validator{});
+    (void) std::search_n(ExecutionPolicy, iarr, iarr, 0, 0, validating_equal{});
+}
+
+void test_parallel_algorithms() {
+    test_per_execution_policy<std::execution::seq>();
+    test_per_execution_policy<std::execution::par>();
+    test_per_execution_policy<std::execution::par_unseq>();
+#if _HAS_CXX20
+    test_per_execution_policy<std::execution::unseq>();
+#endif // _HAS_CXX20
+}
+#endif // _HAS_CXX17
+
+#if _HAS_CXX23 && defined(__cpp_lib_concepts) && !defined(_M_CEE) // TRANSITION, GH-395, VSO-1659496

```

Comment:
`/clr` is *really* incompatible with this incomplete type pattern at this time. We need to guard the entire test (verified internally).